galaxy_api/api/exceptions.py

```diff
@@ -0,0 +1,51 @@
+from django.core.exceptions import PermissionDenied
+from django.http import Http404
+
+from rest_framework import exceptions
+from rest_framework.response import Response
+from rest_framework.settings import api_settings
+
+
+def _get_errors(detail, *, status, title, source=None):
+    if isinstance(detail, list):
+        for item in detail:
+            yield from _get_errors(item, status=status, title=title, source=source)
+    elif isinstance(detail, dict):
+        for key, value in detail.items():
+            yield from _get_errors(value, status=status, title=title, source=key)
+    else:
+        error = {
+            'status': str(status),
+            'code': detail.code,
+            'title': title,
+        }
+
+        if title != detail:
+            error['detail'] = str(detail)
+        if source and source != api_settings.NON_FIELD_ERRORS_KEY:
+            error['source'] = {'parameter': source}
+
+        yield error
+
+
+def exception_handler(exc, context):
+    """Custom exception handler."""
+
+    if isinstance(exc, Http404):
+        exc = exceptions.NotFound()
+    elif isinstance(exc, PermissionDenied):
+        exc = exceptions.PermissionDenied()
+
+    if not isinstance(exc, exceptions.APIException):
+        return None
+
+    headers = {}
+    if getattr(exc, 'auth_header', None):
+        headers['WWW-Authenticate'] = exc.auth_header
+    if getattr(exc, 'wait', None):
+        headers['Retry-After'] = '%d' % exc.wait
+
+    title = exc.__class__.default_detail
+    errors = _get_errors(exc.detail, status=exc.status_code, title=title)
```

Comment:
Exceptions from the DRF project have these `detail` properties, but if any standard exception like a simple `IndexError` is caught, there won't be such properties.

Reply:
In case of any exception that is not expected, e.g. inherited from `rest_framework.exceptions.APIException` or not a Django http exception we follow the behavior of the default exception handler (return `None`).